clients/core-java/src/main/java/org/commonjava/indy/client/core/IndyClientHttp.java

```diff
@@ -607,7 +650,21 @@ public boolean validResponseCode( final int statusCode, final int[] responseCode
     public void close()
     {
         logger.debug( "Shutting down indy client HTTP manager" );
-        connectionManager.reallyShutdown();
+        if ( location != null )
+        {
+            try
+            {
+                factory.close();
```

Comment:
I suggested that because in connect() method on line 152 there is if ( location != null ) return; which means if we don't set location to null it wouldn't be able to reconnect if I understand that correctly. But of course the boolean field closed would work too.